core/src/main/kotlin/io/specmatic/test/DataRecorder.kt

```diff
@@ -0,0 +1,21 @@
+package io.specmatic.test
+
+import io.specmatic.core.log.HttpLogMessage
+
+object DataRecorder {
+
+    val testHttpLogMessages = mutableListOf<HttpLogMessage>()
+    val stubHttpLongMessages = mutableListOf<HttpLogMessage>()
+
+    fun addHttpLog(httpLogMessage: HttpLogMessage) {
+        if(httpLogMessage.scenario != null) {
```

Comment:
This `if` block and null check (along with redundant else) we need to remove / restructure it

Reply:
Thanks, simplified the logic and made it more readable.